cadence-macros/src/state.rs

```diff
@@ -62,10 +62,12 @@ impl<T> SingletonHolder<T> {
     }
 
     /// Set the value if it has not already been set, otherwise this is a no-op
-    #[allow(deprecated)]
     pub fn set(&self, val: T) {
-        // compare_and_swap is deprecated in 1.50 but we target back to 1.36
-        if UNSET != self.state.compare_and_swap(UNSET, LOADING, Ordering::SeqCst) {
+        if self
+            .state
+            .compare_exchange(UNSET, LOADING, Ordering::SeqCst, Ordering::SeqCst)
```

Comment:
for me it looks similar to pattern in https://marabos.nl/atomics/memory-ordering.html#example-locking.

Reply:
I don't remember the reasoning, I probably picked `SeqCst` because it seemed the safest. I'm reading docs now and will open a PR to change it to something more appropriate.